enf-reducer/n-ary-gather-junctors.metta

```diff
@@ -0,0 +1,72 @@
+ (= (concat-tuple $xs $ys)
+    (if (== $xs ()) $ys
+        (let* (($head (car-atom $xs))  
+               ($tail (cdr-atom $xs)) 
+               ($tail-new (concat-tuple $tail $ys))  
+              )
+          (cons-atom $head $tail-new)  
+        )
+    )
+)
+
+(= (fold $f $tuple $acc)
```

Comment:
Can you move it to the `Utilities/general-helper-functions.metta` file. That's where we are going to be keeping functions that deal with any atom.

Reply:
It's also best if you can add a type definition for this function. But don't add the type definition for `$f` as `Atom` it's better to leave it without type definition than give the `$f` a type of `Atom`.

The definition could look something like this:
```scheme
(: fold (-> (-> Atom Atom Atom) Atom Atom)
```

This might sound redundant but it's better in the long run.

Reply:
And also for the `foldl` function, it's not common to put the accumulator in the end. Some one might misunderstand the order of the arguments and cause a problem in the long run. So it's better if you can switch their position to be like so:
```scheme
(= (fold $f $acc $tuple) ...)
```